lib/rouge/lexers/perl6pod.rb

```diff
@@ -0,0 +1,199 @@
+# -*- coding: utf-8 -*- #
+
+module Rouge
+  module Lexers
+    class Pod6 < RegexLexer
+      title "Perl 6 Pod"
+      desc "Perl 6 Plain' Ol Documentation (perl6.org)"
+
+      tag 'perl6pod'
+      aliases 'pod6'
+
+      filenames '*.pod', '*.pod6'
+
+      state :root do
+         rule(/\=begin pod/m, Keyword, :pod6)
+         rule(/^#.*$/, Comment)
+      end
+
+      state :pod6 do
+         rule(/\=end pod/m, Keyword, :pop!)
+
+         rule(/\=NAME/, Keyword, :semantic)
+         rule(/\=AUTHOR/, Keyword, :semantic)
+         rule(/\=VERSION/, Keyword, :semantic)
+         rule(/\=TITLE/, Keyword, :semantic)
+         rule(/\=SUBTITLE/, Keyword, :semantic)
+
+         rule(/\=begin code/, Keyword, :block_code)
+         rule(/\=begin input/, Keyword, :block_input)
+         rule(/\=begin output/, Keyword, :block_output)
+         rule(/\=begin/, Keyword, :block)
+
+         rule(/\=head(\d+)\s+/, Keyword, :head)
+         rule(/\=item(\d+)\s+/, Keyword, :item)
+         rule(/\=input/, Keyword, :input)
+         rule(/\=output/, Keyword, :output)
+         rule(/\=defn/, Keyword)
+
+         rule(/B\</, Punctuation::Indicator, :formatting_b)
+         rule(/C\</, Punctuation::Indicator, :formatting_c)
+         rule(/E\</, Punctuation::Indicator, :formatting_e)
+         rule(/I\</, Punctuation::Indicator, :formatting_i)
+         rule(/K\</, Punctuation::Indicator, :formatting_k)
+         rule(/L\</, Punctuation::Indicator, :formatting_l)
+         rule(/N\</, Punctuation::Indicator, :formatting_n)
+         rule(/T\</, Punctuation::Indicator, :formatting_t)
+         rule(/U\</, Punctuation::Indicator, :formatting_u)
+         rule(/Z\</, Punctuation::Indicator, :formatting_z)
+
+         rule(/^(?:\t|\s{4,})/, Other, :code)
+
+         rule(/./, Text)
+      end
+
+      state :semantic do
+         rule(/.+\n/, Name, :pop!)
+      end
+
+      state :head do
+         rule(/\n/, Text::Whitespace, :pop!)
+
+         rule(/:\w+\</, Name::Attribute, :attribute)
+
+         rule(/./, Generic::Heading)
+      end
+
+      state :item do
+         rule(/\n/, Text::Whitespace, :pop!)
+
+         rule(/B\</, Punctuation::Indicator, :formatting_b)
```

Comment:
The :formatting_... part seems excessive. You can do this with far fewer rules. E.g, something like:

```ruby
    formatting_tokens = { "B" => Generic::String, ... }
    rule(/([BCEIKLN...]<)([^>]*)>/) do m
      t = formatting_tokens[m[0][0]]
      groups Punctuation::Indicator, t , Punctuation::Indicator
    end 
```

.. or similar ought to work. (Untested; let me know if you run into problems)

Reply:
If I use this, I lose the `<` and `>` in the output.